nrf_rpc/nrf_rpc.c

```diff
@@ -597,19 +601,44 @@ static int init_packet_handle(struct header *hdr, const struct nrf_rpc_group **g
 	}
 
 	/* Check for the corresponding local group and initialize it. */
-	*group = remote_group_init(hdr->src_group_id, init_data.strid, init_data.strid_len);
+	*group = group_from_strid(init_data.strid, init_data.strid_len);
 	if (*group == NULL) {
+		NRF_RPC_ERR("Remote group does not match local group");
 		NRF_RPC_ASSERT(false);
 		return -NRF_EFAULT;
 	}
 
-	initialized_group_count++;
-	if (initialized_group_count == group_count) {
+	group_data = (*group)->data;
+	first_init = group_data->dst_group_id == NRF_RPC_ID_UNKNOWN;
+	group_data->dst_group_id = hdr->src_group_id;
+	bool wait_on_init = (*group)->wait_on_init;
```

Comment:
Local variable declarations should be at the beginning.

Reply:
Moved the declaration to the beginning.